### ISSUE TITLE
Dialog appendTo might cause memory leak

### ISSUE THREAD

**Issue by mertsincan** [primefaces/primefaces#367]
If there is appendTo attribute on Dialog, when you update this dialog, a dialog content with same id is added in DOM.

**Comment by mmariotti** [comment]
A Javascript bug has been introduced when searching for existing dialog by id, the value is not escaped (":"), bringing up:  

```
Uncaught Error: Syntax error, unrecognized expression: [id=form:j_idt397:j_idt400]
cv.error                                                                @ jquery.js.jsf?ln=primefaces&v=5.2.5:(25)
ch                                                                      @ jquery.js.jsf?ln=primefaces&v=5.2.5:(25)
dg                                                                      @ jquery.js.jsf?ln=primefaces&v=5.2.5:(25)
cv                                                                      @ jquery.js.jsf?ln=primefaces&v=5.2.5:(25)
bI.fn.extend.find                                                       @ jquery.js.jsf?ln=primefaces&v=5.2.5:(25)
bI.fn.init                                                              @ jquery.js.jsf?ln=primefaces&v=5.2.5:(25)
bI                                                                      @ jquery.js.jsf?ln=primefaces&v=5.2.5:(14)
PrimeFaces.widget.Dialog.PrimeFaces.widget.BaseWidget.extend.refresh    @ primefaces.js.jsf?ln=primefaces&v=5.2.5:(19)  --->    if(a.appendTo){var b=$("[id="+a.id+"]");
b.createWidget                                                          @ primefaces.js.jsf?ln=primefaces&v=5.2.5:(1)
b.cw                                                                    @ primefaces.js.jsf?ln=primefaces&v=5.2.5:(1)
(anonymous function)                                                    @ VM4959:1                                      --->    $(function(){PrimeFaces.cw("Dialog","form_j_idt397_dialog",{id:"form:j_idt397:j_idt400",widgetVar:"form_j_idt397_dialog",resizable:false,modal:true,appendTo:"form:j_idt397",dynamic:true,onHide:function(){unloadDialog(this)}});});
```

**Comment by mertsincan** [comment]
Replicated the issue. I created a new issue. (#393)

**Comment by cagataycivici** [comment]
Yes, there is an ugly bug fixed in #393.
